### ISSUE TITLE
Add flag --install-exit-handlers for GraalVM

### ISSUE THREAD

**Issue by ilopmar** [micronaut-projects/micronaut-core#3880]
GraalVM 20.1.0 introduced the flag `--install-exit-handlers`. See https://www.graalvm.org/docs/release-notes/20_1/#native-image

> Added new option for the image generator to exit handlers on termination signals: `--install-exit-handlers`. This improves the user experience when running a native image in a Docker container as the init process, since Linux does not install a default signal handler for the process with id “1” and the native image is therefore not reacting to a shutdown request from Docker.

I tried this sometime ago but it failed with `--static` images. We don't create static native images anymore because it is discouraged and it has issues with dns calls and other network calls. It is possible to use compile static against `muslc` library instead glibc, but it only works for JDK11, so at this moment this is not an option for us.

Add the flag `--install-exit-handlers` to runtime module.

**Comment by ilopmar** [comment]
I've reverted the change because it breaks AWS Lambda native-image support. See https://github.com/oracle/graal/issues/2742

**Comment by ilopmar** [comment]
The [GraalVM issue #2742](https://github.com/oracle/graal/issues/2742) has been fixed but the fix will be included in 20.3 (to be released on November 17th 2020). Until that date we can't merge the fix.

In the mean time, to avoid regressions I've manually enabled the flag in `basic-app`, `aws-app` and `function-aws` test applications. [This pipeline ](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/pipelines/177001825) passes with the flag enabled in those applications (check the commit messages). Once the fix is merged we can remove the flag from the test applications.
